### PR TITLE
Tas interface commands

### LINKED ISSUE #214: Timestamp
We should extend the interface with a generated timestamp.

While we are at, we can also include a generator version and a mod name.

### PR DESCRIPTION
**Pull request by theis999**
Added the TAS console commands to the interface, allowing other mods to use them.
Which is useful for my TAS helper, as i can create buttons that execute the commands.

Control.lua scrips of "Theis TAS helper" where it is used:
[control.zip](https://github.com/MortenTobiasNielsen/Factorio-TAS-Generator/files/11086984/control.zip)

Note: this is built on #215 to prevent merge conflicts

Closes #214

### CODE REVIEW

File: src/GenerateScript.cpp [hunk 1]
```diff
@@ -18,15 +21,47 @@ void GenerateScript::reset()
...
 	total_steps = 1;
-	step_list = "local step = {}\n\n";
+	step_list = "";
+	std::stringstream ss (step_list);
+	ss << endl << "local tas_generator = {" << endl
+		<< tab << "name = \"" << generator_thumbprint.name << "\"" << comma_endl
+		<< tab << "version = \"" << generator_thumbprint.version << "\"" << comma_endl
+		<< tab << "tas = {" << endl
+		<< tab << tab << "name = \"" << name << "\"" << comma_endl
+		<< tab << tab << "timestamp = \"" << currentDateTime() << "\"" << comma_endl
```
Comment (MortenTobiasNielsen) [inline src/GenerateScript.cpp:51]: I would like to use std::chrono::system_clock::now(); unless you really need the very specific date format made in the currentDateTime function.

File: src/GenerateScript.cpp [hunk 2]
```diff
@@ -64,6 +99,7 @@ void GenerateScript::PaintWalk(string step, bool paint)
 
 void GenerateScript::generate(wxWindow* parent, DialogProgressBar* dialog_progress_bar, vector<StepParameters> steps, string& folder_location, bool auto_close, bool only_generate_script, string goal)
 {
+	this->name = folder_location.substr(folder_location.find_last_of('\\') + 1);
```
Comment (MortenTobiasNielsen) [inline src/GenerateScript.cpp:102]: Is this used for anything?

**Review by MortenTobiasNielsen — CHANGES_REQUESTED** [review]

File: src/GenerateScript.cpp (hunk 2 above)
Comment (theis999) [reply src/GenerateScript.cpp:102]: It is used exactly once, in the method GenerateScript::ClearSteps() line 50.

File: src/GenerateScript.cpp (hunk 1 above)
Comment (theis999) [reply src/GenerateScript.cpp]: Done

**Review by MortenTobiasNielsen — APPROVED** [review]

**Pull request merged**